commit 9f6a404acb445f75a8ace55439a3401c698b6e5d
Author: Matthias Reichl <hias@horus.com>
Date:   Fri Feb 2 20:30:41 2018 +0100

    ASoC: pcm512x: revert downstream changes
    
    This partially reverts commit 185ea05465aac8bf02a0d2b2f4289d42c72870b7
    which was added by https://github.com/raspberrypi/linux/pull/1152
    
    The downstream pcm512x changes caused a regression, it broke normal
    use of the 24bit format with the codec, eg when using simple-audio-card.
    
    The actual bug with 24bit playback is the incorrect usage
    of physical_width in various drivers in the downstream tree
    which causes 24bit data to be transmitted with 32 clock
    cycles. So it's not the pcm512x that needs fixing, it's the
    soundcard drivers.
    
    Signed-off-by: Matthias Reichl <hias@horus.com>

diff --git a/sound/soc/codecs/pcm512x.c b/sound/soc/codecs/pcm512x.c
index 69b3fae6c4ed..e0f3556d3872 100644
--- a/sound/soc/codecs/pcm512x.c
+++ b/sound/soc/codecs/pcm512x.c
@@ -851,8 +851,7 @@ static int pcm512x_set_dividers(struct snd_soc_dai *dai,
 	int fssp;
 	int gpio;
 
-	lrclk_div = snd_pcm_format_physical_width(params_format(params))
-		* params_channels(params);
+	lrclk_div = snd_soc_params_to_frame_size(params);
 	if (lrclk_div == 0) {
 		dev_err(dev, "No LRCLK?\n");
 		return -EINVAL;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1187_9f6a404acb445f75a8ace55439a3401c698b6e5d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1187_9f6a404acb445f75a8ace55439a3401c698b6e5d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
